tests/chain.js

```diff
@@ -18,10 +18,41 @@ metasync.for([1, 2, 3, 4]).filter((item, cb) => {
 }).reduce((a, b, cb) => {
   process.nextTick(cb, null, a + b);
 }).then((result) => {
-  console.log('Chaining test done: ' + result);
-  assert.strictEqual(result, 12);  // 2 * 2 + 4 * 2
+  console.log('Chaining test #1 with process.nextTick done: ' + result);
+  assert.strictEqual(result, 12); // 2 * 2 + 4 * 2
 }).catch((error) => {
   const description = error.stack || 'Error: ' + error.toString();
   console.error(description);
   process.exit(1);
 });
+
+metasync.for([1, 2, 3, 4]).filter((item, cb) => {
```

Comment:
Yeah, it is crucial that we test both callbacks invoked on the same event loop tick and on next ticks.